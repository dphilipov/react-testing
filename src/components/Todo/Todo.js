import React from 'react';

function Todo({ todo }) {
  const { id, title, completed } = todo;

  return (
    <div data-testid="test-1">
      {
        completed
          ? <strike>< h2 >
            {title}
          </h2 ></strike>
          : < h2 >
            {title}
          </h2 >
      }
    </div >
  )
}

export default Todo;
