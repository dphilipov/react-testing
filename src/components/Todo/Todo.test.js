import { render, screen, cleanup, } from '@testing-library/react';
import Todo from './Todo';

afterEach(() => {
    cleanup();
})

test('should render uncompleted Todo component', () => {
    const todo = { id: 1, title: 'Test title', completed: false };

    render(<Todo todo={todo} />);

    const TodoElement = screen.getByTestId('test-1');

    expect(TodoElement).toBeInTheDocument();
    expect(TodoElement).not.toContainHTML('<strike>');
})

test('should render uncompleted Todo component', () => {
    const todo = { id: 2, title: 'Test title 2', completed: true };

    render(<Todo todo={todo} />);

    const TodoElement = screen.getByTestId('test-2');

    expect(TodoElement).toBeInTheDocument();
    expect(TodoElement).toMatchSnapshot('<strike>');
})