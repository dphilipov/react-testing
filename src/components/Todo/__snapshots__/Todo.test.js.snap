// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render uncompleted Todo component: <strike> 1`] = `
<div
  data-testid="test-2"
>
  <strike>
    <h2>
      Test title 2
    </h2>
  </strike>
</div>
`;
